openmc/weight_windows.py

```diff
@@ -941,13 +767,282 @@ def hdf5_to_wws(path='weight_windows.h5'):
+
+        if nwg == 1:
+            mesh = RectilinearMesh()
+            mesh.x_grid, mesh.y_grid, mesh.z_grid = grids
+        elif nwg == 2:
+            mesh = CylindricalMesh(
+                r_grid=grids[0],
+                z_grid=grids[1],
+                phi_grid=grids[2],
+                origin = xyz0,
+            )
+        elif nwg == 3:
+            mesh = SphericalMesh(
+                r_grid=grids[0],
+                theta_grid=grids[1],
+                phi_grid=grids[2],
+                origin = xyz0
+            )
+
+        # extract weight window values from array
+        wws = cls()
+        for ne_i, nt_i, particle_type in zip(ne, nt, ('neutron', 'photon')):
+            # no information to read for this particle if
+            # either the energy bins or time bins are empty
+            if ne_i == 0 or nt_i == 0:
+                continue
+
+            if iv > 1:
+                # time bins are parsed but unused for now
+                end_idx = start_idx + nt_i
+                time_bounds = ww_data[start_idx:end_idx]
+                np.insert(time_bounds, (0,), (0.0,))
+                start_idx = end_idx
+
+            # read energy boundaries
+            end_idx = start_idx + ne_i
+            energy_bounds = np.insert(ww_data[start_idx:end_idx], (0,), (0.0,))
+            # convert from MeV to eV
+            energy_bounds *= 1e6
+            start_idx = end_idx
+
+            # read weight window values
+            end_idx = start_idx + (nfx * nfy * nfz) * nt_i * ne_i
+
+            # read values and reshape according to ordering
+            # slowest to fastest: t, e, z, y, x
+            # reorder with transpose since our ordering is x, y, z, e, t
+            ww_shape = (nt_i, ne_i, nfz, nfy, nfx)
+            ww_values = ww_data[start_idx:end_idx].reshape(ww_shape).T
+            # Only use first time bin since we don't support time dependent weight
+            # windows yet.
+            ww_values = ww_values[:, :, :, :, 0]
+            start_idx = end_idx
+
+            # create a weight window object
+            ww = WeightWindows(id=None,
+                            mesh=mesh,
+                            lower_ww_bounds=ww_values,
+                            upper_bound_ratio=5.0,
+                            energy_bounds=energy_bounds,
+                            particle_type=particle_type)
+            wws.append(ww)
+
+        return wws
+
+    def export_to_hdf5(self, path: PathLike = 'weight_windows.h5', **init_kwargs):
+        """Write weight windows to an HDF5 file.
+
+        Parameters
+        ----------
+        path : PathLike
+            Path to the file to write weight windows to
+        **init_kwargs
+            Keyword arguments passed to :func:`openmc.lib.init`
+
+        """
+        import openmc.lib
+        cv.check_type('path', path, PathLike)
+
+        # Create a temporary model with the weight windows
+        model = openmc.Model()
```

Comment:
Not requesting this here necessarily, but given that we seem to be using what I'll call minimal proxy models to access `openmc.lib` functionality more and more often, it might be good to add a `minimal_model` to `openmc.Model` at some point.

Reply:
For sure. If it starts cropping up elsewhere we might want to consider it.